bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx

```diff
@@ -0,0 +1,412 @@
+     uploading any office document or PDF file. We recommend for the best results,
+     to please upload a PDF file.`,
+  },
+  confirmLabel: {
+    id: 'app.presentationUploder.confirmLabel',
+    defaultMessage: 'Start',
+  },
+  confirmDesc: {
+    id: 'app.presentationUploder.confirmDesc',
+    defaultMessage: 'Save your changes and start the presentation',
+  },
+  dismissLabel: {
+    id: 'app.presentationUploder.dismissLabel',
+    defaultMessage: 'Cancel',
+  },
+  dismissDesc: {
+    id: 'app.presentationUploder.dismissDesc',
+    defaultMessage: 'Closes and discarts your changes',
+  },
+  dropzoneLabel: {
+    id: 'app.presentationUploder.dropzoneLabel',
+    defaultMessage: 'Drag files here to upload',
+  },
+  browseFilesLabel: {
+    id: 'app.presentationUploder.browseFilesLabel',
+    defaultMessage: 'or browse for files',
+  },
+  fileToUpload: {
+    id: 'app.presentationUploder.fileToUpload',
+    defaultMessage: 'To be uploaded...',
+  },
+  uploadProcess: {
+    id: 'app.presentationUploder.upload.progress',
+    defaultMessage: 'Uploading ({progress}%)',
+  },
+  413: {
+    id: 'app.presentationUploder.upload.413',
+    defaultMessage: 'File is too large.',
+  },
+  conversionProcessingSlides: {
+    id: 'app.presentationUploder.conversion.conversionProcessingSlides',
+    defaultMessage: 'Processing page {current} of {total}',
+  },
+  genericConversionStatus: {
+    id: 'app.presentationUploder.conversion.genericConversionStatus',
+    defaultMessage: 'Converting file...',
+  },
+  GENERATING_THUMBNAIL: {
+    id: 'app.presentationUploder.conversion.generatingThumbnail',
+    defaultMessage: 'Generating thumbnails...',
+  },
+  GENERATED_SLIDE: {
+    id: 'app.presentationUploder.conversion.generatedSlides',
+    defaultMessage: 'Slides generated...',
+  },
+});
+
+const isProcessingOrUploading = item => item && (!item.conversion.done || !item.upload.done);
+
+function updateFileUploadState(id, state) {
+  this.setState(({ presentations }) => {
+    const fileIndex = presentations.findIndex(f => f.id === id);
+    return {
+      presentations: update(presentations, {
+        [fileIndex]: { $apply: file =>
+          update(file, {
+            upload: { $apply: upload => update(upload, { $merge: state }) },
+          }),
+        },
+      }),
+    };
+  });
+}
+
+class PresentationUploader extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      presentations: props.presentations,
+      preventClosing: false,
+      disableActions: false,
+    };
+
+    this.handleConfirm = this.handleConfirm.bind(this);
+    this.handleDismiss = this.handleDismiss.bind(this);
+    this.handleFiledrop = this.handleFiledrop.bind(this);
+    this.handleCurrentChange = this.handleCurrentChange.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const presentationStateUpdated =
+      this.state.presentations.map(p =>
```

Comment:
Ideally we would be relying on a different method for finding unique presentation file. Now a/test.pdf and b/test.pdf will have the same filename (path dropped) and will cause a conflict